tutorials/dapt-curation/README.md

```diff
@@ -0,0 +1,67 @@
+# Data curation for DAPT (Domain Adaptive Pre-Training)
+
+[ChipNeMo](https://arxiv.org/pdf/2311.00176) is a chip design domain adapted LLM. LLama models are continually pre-trained with 20B plus tokens on domain-specific chip design data, including code, documents, etc., based on NeMo foundation models and then fine-tuned with instruction datasets from design data as well as external sources.
+
+Here, we share a tutotial with best practices on data curation for DAPT (domain-adaptive pre-training) for a ChipNeMo-like code generation use case.
+
+* In this tutorial, we will leverage chip domain/hardware datasets from open-source GitHub repositories (`.code/sources/github_repos.jsonl`), wiki URLs (`.code/sources/wikipedia_urls.jsonl`), and academic papers(`./sources/arxiv_urls.jsonl`).
+
+* `\code\data` contains curated data after processing
+
+* Notebook for data curation with NeMo Curator: `\code\DAPT_data_curation.ipynb`
+
+The small size of this dataset makes it ideal for creating and validating data curation pipelines on a local machine or a cluster.
+
+This playbook utilizes specific tools and techniques. First, we convert all files to Txt format (if not already in Txt), compress files on disk, add metadata, and convert them to JSON (`./data/raw/`). Then, we leverage [NeMo Curator](https://github.com/NVIDIA/NeMo-Curator/tree/main) to mine high-quality text at scale from a massive code-generation corpus. [NeMo Curator](https://github.com/NVIDIA/NeMo-Curator/tree/main), built on Dask and RAPIDS, is instrumental in scaling data curation and providing GPU acceleration. We use its capabilities to extract text, identify code file types, fix unicode errors, filter quality through heuristics, deduplicate, and redact personal information. We finally also provide steps to blend and shuffle data sources for continued pre-training.
+
+
+## NeMo Tools and Resources
+
+* [NeMo-Curator GitHub repo](https://github.com/NVIDIA/NeMo-Curator/tree/main)
+
+
+## Software Requirements
+* This playbook has been tested on: nvcr.io/nvidia/nemo:24.03.01.framework. It is expected to work similarly on other environments. Within the container, from the NeMo-Curator directory install using `pip install --extra-index-url https://pypi.nvidia.com ".[cuda12x]"`
```

Comment:
Let's add a note that this can be executed on a local machine with a GPU, but it has been tested with .....
Same as above, let's remove the install instructions.